bundles/core/src/main/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClient.java

```diff
@@ -217,6 +221,28 @@ private MagentoGraphqlClient(Resource resource, Page page, SlingHttpServletReque
         }
     }
 
+    private List<Header> getCustomHttpHeaders(ComponentsConfiguration configuration) {
+        List<Header> headers = new ArrayList<>();
```

Comment:
you don't need to instantiate the ArrayList rather use `Collections.emptyList()`

Reply:
Isn't it the same thing?

Reply:
`Collections.emptyList()` returns a singleton instancce so you save the unneeded allocation of a new ArrayList

Reply:
It's true, but to the list returned here, we add more headers in the constructor. To be 100% "clean" we could probably return an `emptyList()` here, but use `new ArrayList<>(this list)` [here](https://github.com/adobe/aem-core-cif-components/blob/features/CIF-2097/bundles/core/src/main/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClient.java#L181), which looks like a bit of an overkill to me

Reply:
true, totally valid for a private method 👍